backend/src/clients/mistral-client/index.ts

```diff
@@ -0,0 +1,459 @@
+import { config } from "../../config/config";
+import logger from "../../config/winston";
+import {
+  Document,
+  FileData,
+  LanguageRequest,
+  PerspectiveRequest,
+  Prompt,
+  TagRequest,
+} from "../../types/interfaces";
+import { BackendLLMClient } from "../llm-client/llmClientMapping";
+import { UploadedFile } from "express-fileupload";
+import TaskQueue from "../../utility/jobQueue";
+import { languagePrompt } from "../../utility/prompts";
+import { IpfsClient } from "../ipfs-client";
+import pdf from "pdf-parse";
+import mammoth from "mammoth";
+import { Mistral } from "@mistralai/mistralai";
+
+export class MistralClient implements BackendLLMClient {
+  private mistral: Mistral;
+
+  constructor() {
+    this.mistral = new Mistral({
+      apiKey: config.mistralApiKey,
+    });
+  }
+
+  async generateCompletion(
+    fileData: FileData,
+    prompt: string,
+    title: string
+  ): Promise<{ title: string; summary: string }> {
+    try {
+      const systemPrompt = `${prompt}\n\nHere is the document content:\n\n${fileData.data.content}`;
+
+      const chatCompletion = await this.mistral.chat.complete({
+        model: config.mistralModel,
+        messages: [
+          {
+            role: "system",
+            content: "You are a helpful assistant that summarizes documents.",
+          },
+          { role: "user", content: systemPrompt },
+        ],
+      });
+
+      const content = chatCompletion.choices[0].message.content;
+      const summary = Array.isArray(content)
+        ? content
+            .filter((c) => c.type === "text")
+            .map((c) => c.text)
+            .join("")
+        : (content ?? "");
+      logger.debug(
+        `Mistral chatCompletionResult\n${JSON.stringify(chatCompletion, null, 2).slice(0, 280)}...`
+      );
+
+      return { title, summary };
+    } catch (error) {
+      logger.error(
```

Comment:
please rewrite all logger.error to look like this `logger.error("text", e);` so we can see the whole error stack. Stringifying or putting the message and error in `` prints out just the message from the error and leaves out the error stack